src/libs/ViewportDetector.js

```diff
@@ -0,0 +1,104 @@
+/**
+ * Copyright 2015, Yahoo Inc.
+ * Copyrights licensed under the New BSD License. See the accompanying LICENSE file for terms.
```

Comment:
this file is duplicated with one of our internal implementations, it makes sense to be centralized. I will do that later. for now keep that here to make this PR cleaner.